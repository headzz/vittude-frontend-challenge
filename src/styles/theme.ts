export const theme = {
  colors: {
    primary: '#FF6952',
    secondary: '#F26B5B',
    error: '#F63D15',
    dark: {
      100: '#777C88',
      200: '#2E3035',
      300: '#000000',
    },
    light: {
      100: '#FFFFFF',
      200: '#E7EAF4',
    },
  },
  font: {
    size: {
      normal: '1rem',
      small: '0.875rem',
      large: '2rem',
      extraLarge: '3rem',
    },
    type: { header: 'Inter', normal: 'Fira Sans' },
  },
  spacing: {
    extraSmall: '10px',
    small: '12px',
    normal: '16px',
    large: '24px',
    extraLarge: '48px',
  },
}
