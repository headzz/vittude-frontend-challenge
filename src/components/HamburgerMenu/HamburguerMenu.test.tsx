import { fireEvent, render, screen } from '@testing-library/react'
import { HamburguerMenu } from '.'
import { ThemeProvider } from 'styled-components'
import { theme } from '@styles/theme'

describe('hamburguer mennus', () => {
  it('renders default menu', () => {
    render(
      <ThemeProvider theme={theme}>
        <HamburguerMenu>
          <ul>
            <li>Inicio</li>
            <li>Blog</li>
          </ul>
        </HamburguerMenu>
      </ThemeProvider>,
    )

    const button = screen.getByRole('button')

    expect(button).toBeInTheDocument()
    expect(button).toMatchSnapshot()
  })

  it('should trigger nav appearing', () => {
    render(
      <ThemeProvider theme={theme}>
        <HamburguerMenu>
          <ul>
            <li>Inicio</li>
            <li>Blog</li>
          </ul>
        </HamburguerMenu>
      </ThemeProvider>,
    )

    const button = screen.getByRole('button')
    fireEvent.click(button)
    const nav = screen.getByRole('navigation')

    expect(nav).toBeInTheDocument()
    expect(screen.getByText(/blog/i)).toBeInTheDocument()
    expect(nav).toMatchSnapshot()
  })
})
