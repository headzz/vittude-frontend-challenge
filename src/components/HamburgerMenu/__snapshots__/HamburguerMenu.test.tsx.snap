// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders hamburguer menu 1`] = `
<button
  class="sc-bcPKhP heUomK"
>
  <span
    class="sc-grXZZQ irOzWR"
  />
  <span
    class="sc-grXZZQ irOzWR"
  />
</button>
`;

exports[`Button should trigger nav appearing 1`] = `
<nav
  class="sc-dkjKgF iKmdmO"
  open=""
>
  <ul>
    <li>
      Inicio
    </li>
    <li>
      Blog
    </li>
  </ul>
</nav>
`;
